### DISCUSSION TITLE
Trojan Websocket

### DISCUSSION THREAD

**Discussion by Stevemoretz** [FranzKafkaYu/x-ui#241] [Q&A]
Hi I see that in this panel there is a websocket transport option available in the UI

![image](https://user-images.githubusercontent.com/27680142/212532012-413274e4-5f67-4fa7-bc9d-384b2a183518.png)

Is that just a UI bug? Because I remember the creator of trojan protocol saying that this protocol can't be put behind a cdn for instance cloudflare but if it does support websocket that means that it can.

Also in all similar panels that I've ever tried there is no such option for trojan to select websocket for it, could you please explain what's going on?

Thank you

**Comment by FranzKafkaYu** [comment]
@Stevemoretz No，xray core has made some changes on Trojan Protocol，actually it do support WebSocket or grpc etc,you can have a test.
